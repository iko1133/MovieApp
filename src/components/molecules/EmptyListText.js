import React from "react";
import styled from "styled-components/native";

const Container = styled.View`
  flex: 1;
`;

const EmptyListLabel = styled.Text`
  align-self: center;
  font-size: 16px;
`;

const EmptyListText = ({ text = "List is empty!" }) => {
  return (
    <Container>
      <EmptyListLabel>{text}</EmptyListLabel>
    </Container>
  );
};

export default EmptyListText;
