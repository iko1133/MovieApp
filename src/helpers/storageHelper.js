import AsyncStorage from "@react-native-async-storage/async-storage";

/**
 * A function that will save given object under given key
 * @param {object} object - An object that needs to be saved in storage
 * @param {string} key - A string key the object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 */
export const storeObject = async (
  object,
  key,
  errorText = "There was an error while trying to save"
) => {
  try {
    const jsonObj = JSON.stringify(object);
    await AsyncStorage.setItem(key, jsonObj);
  } catch (e) {
    alert(errorText);
  }
};

/**
 * A function that will return an object saved under given key or null if none was found
 * @param {string} key - A string key that desired object should be saved under
 * @param {string} errorText - A string that will be displayed if something goes wrong
 * @returns Object under given key or null if none was found
 */
export const getObject = async (key, errorText) => {
  try {
    const jsonVal = await AsyncStorage.getItem(key);
    return jsonVal != null ? JSON.parse(jsonVal) : null;
  } catch (e) {
    alert(errorText);
  }
};
