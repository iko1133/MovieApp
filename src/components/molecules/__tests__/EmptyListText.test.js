import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import EmptyListText from "../EmptyListText";

describe("Testing EmptyListText", () => {
  it("Renders EmptyListText", () =>
    expect(
      renderer.create(WrapInThemeProvider(<EmptyListText />))
    ).toMatchSnapshot());

  it("Renders default empty list text", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <EmptyListText text="NEW, UPDATED, LONGER TEXT THAT SHOULD BE SCREEN SIZE" />
        )
      )
    ).toMatchSnapshot());
});
