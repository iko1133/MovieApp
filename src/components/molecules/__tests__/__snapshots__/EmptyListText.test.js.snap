// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EmptyListText Renders EmptyListText 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 16,
        },
      ]
    }
  >
    List is empty!
  </Text>
</View>
`;

exports[`Testing EmptyListText Renders default empty list text 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontSize": 16,
        },
      ]
    }
  >
    NEW, UPDATED, LONGER TEXT THAT SHOULD BE SCREEN SIZE
  </Text>
</View>
`;
